html/datasets.html

```diff
@@ -0,0 +1,310 @@
+                        if (record["citations"].length == 0) {
+                            html += 'N/A'
+                        } else {
+                            for (var i = 0; i < record["citations"].length; i++) {
+                                html += "<a href=\"" + record["citations"][i]["uri"] + "\" target=\"_blank\">" + record["citations"][i]["name"] + "</a>";
+                                if (i < record["citations"].length - 1) {
+                                    html += "; ";
+                                }
+                            }
+                        }
+                        html += "</small></span>";
+                    }
+
+                    // display distributions
+                    if (record["distributions"]) {
+                        html += "<br /><span><small>Distributions: ";
+                        if (record["distributions"].length == 0) {
+                            html += 'N/A'
+                        } else {
+                            html += "<br /> &emsp;&emsp;&emsp;&emsp;";
+                            for (var i = 0; i < record["distributions"].length; i++) {
+                                html += ("<a href=\"" + record["distributions"][i]["accessURL"] + "\" target=\"_blank\">"
+                                        + record["distributions"][i]["name"] + "</a>");
+                                html += " (Direct access: ";
+                                for (var j = 0; j < record["distributions"][i]["files"].length; j++){
+                                    html += ("<a href=\"" + record["distributions"][i]["files"][j]["downloadURL"]
+                                            + "\" target=\"_blank\">"
+                                            + record["distributions"][i]["files"][j]["name"] + "</a>");
+                                    if (j < record["distributions"][i]["files"].length - 1) {
+                                        html += "; ";
+                                    }
+                                }
+                                html += ")";
+                                if (i < record["distributions"].length - 1) {
+                                    html += "; <br /> &emsp;&emsp;&emsp;&emsp;";
+                                }
+                            }
+                        }
+                        html += "</small></span>";
+                    }
+
+                    // Badges
+
+                    html += "<br />";
+
+                    if (record["doi"]) {
+                        var escapedDOI = encodeURIComponent(record["doi"]);
+                        html += "<a href=\"" + doiUrl + "\" target=\"_blank\"><img style='vertical-align: middle' src='//dcotest.tw.rpi.edu/badges/badge?title=DOI&text="+escapedDOI+"&color=66CCFF'/></a>&emsp;";
+                    }
+
+                    var escapedDCOId= encodeURIComponent(record["dcoId"].substring(25));
```

Comment:
Above you check to see if the dcoid exists. If it doesn't then this line breaks because you get back a null object and then try to use substring on it. Found this problem already in the people browser.